### DISCUSSION TITLE
Am I understanding targets wrongly?

### DISCUSSION THREAD

**Discussion by azapater** [theory/sqitch#657] [Q&A] [answered]
I'm using firebird 3 and I have 2 dbs (let's call them dev and prod) I need to keep in different versions. My initial idea was to have a common project and plan for both of them, and use dev as the default one. Once I tag a version, then manually deploy to prod to the specific tag I need using the command:  

`sqitch deploy --to XX --target prod`

Note: Both databases needed to be added as `--log-only`. I followed this tutorial succcessfully for dev, creating the first change with the schema of the whole db.  

https://github.com/sqitchers/sqitch/wiki/How-do-I-get-started-with-an-existing-database%3F

The problem I'm facing is that I can't add the db prod as `--log-only`. Sqitch says that there is nothing to deploy and that the db is up to date, even though I've never deployed anything to prod. If I check the status of prod it says that is up-to-date showing the last change as deployed. 

I've tried to remove the prod target and add it again using the cli with the same result. I've checked the database created (sqitch.fdb) and in the table PROJECTS, there is just one created for dev, but not one for prod. Maybe this could be the problem.  

I'm not sure if I'm understanding the target concept wrongly, I've done some mistake creating the targets or there is a bug. 

Thanks in advance

**Comment by azapater** [comment, marked as answer]
I managed to find the solution for this situation. I answer my own question, just in case is useful for anyone in the future.

To acomplish this, you need to define independent registry databases for each target. Sqitch can't handle the registry of the same project in different databases, so you just need a diferent registry db for each target. 

Example:

```
[target "dev"]
	uri = db:firebird://xxxxxxxxxx
	registry = dev-sqitch
[target "prod"]
	uri = db:firebird://xxxxxxxxxx
	registry = prod-sqitch
```

**Reply by theory** [reply]
Yes, this is very much a pattern for the database engines where the registry is stored in a separate database, including MySQL, Firebird, and SQLite. For other engines Sqitch defaults to a separate schema in the same database, which is a bit cleaner and less surprising. At one time there was a request for an option to use a prefix in the same database, but it has not yet been finished (see #247 for details).
